test/suites/drift/suite_test.go

```diff
@@ -335,5 +337,62 @@ var _ = Describe("Drift", Label("AWS"), func() {
 			}}),
 			Entry("Start-up Taints Drift", "Start-up Taint", test.ProvisionerOptions{StartupTaints: []v1.Taint{{Key: "example.com/another-taint-2", Effect: v1.TaintEffectPreferNoSchedule}}}),
 		)
+		DescribeTable("awsnodetemplate drift", func(fieldName string, awsnodetemplatespec v1alpha1.AWSNodeTemplateSpec) {
+			if fieldName == "InstanceProfile" {
+				By("creating an instance profile")
+				instanceProfileName := awssdk.String("instance-profile-drift")
+				createInstanceProfile := &iam.CreateInstanceProfileInput{
```

Comment:
Does this get cleaned up? It seems like this will get leaked. How do we make sure that we clean-up this instance profile in the standard E2E test